database/database_handler.py

```diff
@@ -111,3 +93,48 @@ def managed_session(self):
             raise e
         finally:
             session.close()
+
+    def close_connection(self):
+        try:
+            # Dispose the engine
+            self.engine.dispose()
+        except SQLAlchemyError as e:
+            LOGGER.info(f"Error disposing the database engine: {str(e)}")
+            raise e
+
+    @staticmethod
+    def extract_upload_date(path):
```

Comment:
This doesn't seem the right place to this function. Database handler should handle database connection and queries. This is more project related. Especially if in the future we want to extract the database handler to its own repo.

Reply:
Move it to val.py? Or make a new "parse.py" class in the database folder?